### PR TITLE
block app edit and token generation for inactive application

### PR DESCRIPTION
**Pull request by chamilaadhi**
### Proposed changes in this pull request

[List all changes you want to add here. If you fixed an issue, please
add a reference to that issue as well.]

- block token generation and app edit for inactive apps from rest api
- disable token generation and app edit for inactive/rejected apps from UI
### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] ** Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] ** Are all UI and API inputs run through forms or serializers?**
- [ ] ** Are all external inputs validated and sanitized appropriately?**
- [ ] ** Does all branching logic have a default case?**
- [ ] ** Does this solution handle outliers and edge cases gracefully?**
- [ ] ** Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/web/apps/org.wso2.carbon.apimgt.store/src/main/fragments/application-navbar/application-navbar.hbs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{{#if @params.isApplicationActive}}
+<li>
+     <a title="Edit Applications" href="{{@contextPath}}/applications/{{@params.applicationId}}/edit">
+         <span class="icon fw-stack">
+            <i class="fw fw-edit fw-stack-1x"></i>
+            <i class="fw fw-circle-outline fw-stack-2x"></i>
+         </span>
+        Edit
```
Comment (anuruddhal) [inline components/web/apps/org.wso2.carbon.apimgt.store/src/main/fragments/application-navbar/application-navbar.hbs]: Make this i18n

File: components/web/apps/org.wso2.carbon.apimgt.store/src/main/fragments/application-details/application-details.hbs [hunk 2]
```diff
@@ -6,9 +6,11 @@
 <div class="white-wrapper">
     <ul class="nav nav-tabs" role="tablist" id="tabs">
         <li role="presentation" id="details-tab" class="active"><a title="Details" href="#appdetails" aria-controls="profile" role="tab" data-toggle="tab">Details</a></li>
```
Comment (anuruddhal) [inline components/web/apps/org.wso2.carbon.apimgt.store/src/main/fragments/application-details/application-details.hbs]: Please make the texts i18n.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/ApplicationsApiServiceImpl.java [hunk 3]
```diff
@@ -210,6 +211,16 @@ public Response applicationsApplicationIdPut(String applicationId, ApplicationDT
             }
 
             Application application = ApplicationMappingUtil.fromDTOtoApplication(body, username);
+            if (!ApplicationStatus.APPLICATION_APPROVED.equals(application.getStatus())) {
+                String errorMessage = "Application " + applicationId + " is not active";
+                ExceptionCodes exceptionCode = ExceptionCodes.APPLICATION_INACTIVE;
+                APIManagementException e = new APIManagementException(errorMessage, exceptionCode);
+                HashMap<String, String> paramList = new HashMap<String, String>();
+                ErrorDTO errorDTO = RestApiUtil.getErrorDTO(e.getErrorHandler(), paramList);
```
Comment (fazlan-nazeem) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/ApplicationsApiServiceImpl.java]: an empty paramList is passed. We can add the application id to the paramList I guess.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/ApplicationsApiServiceImpl.java [hunk 4]
```diff
@@ -250,6 +261,16 @@ public Response applicationsGenerateKeysPost(String applicationId, ApplicationKe
         try {
             APIStore apiConsumer = RestApiUtil.getConsumer(username);
             Application application = apiConsumer.getApplication(applicationId, username, null);
+            if (application != null && !ApplicationStatus.APPLICATION_APPROVED.equals(application.getStatus())) {
+                String errorMessage = "Application " + applicationId + " is not active";
+                ExceptionCodes exceptionCode = ExceptionCodes.APPLICATION_INACTIVE;
+                APIManagementException e = new APIManagementException(errorMessage, exceptionCode);
+                HashMap<String, String> paramList = new HashMap<String, String>();
+                ErrorDTO errorDTO = RestApiUtil.getErrorDTO(e.getErrorHandler(), paramList);
```
Comment (fazlan-nazeem) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/ApplicationsApiServiceImpl.java]: Same as above comment

**Pull request merged**